synapse/replication/tcp/streams/_base.py

```diff
@@ -299,13 +321,15 @@ class PushRulesStream(Stream):
 
     def __init__(self, hs):
         self.store = hs.get_datastore()
-        super(PushRulesStream, self).__init__(hs)
+        super(PushRulesStream, self).__init__(
+            self._current_token, self._update_function
```

Comment:
`self._update_function` doesn't seem to return an awaitable in this case (although it is `awaitable` itself). Not sure if this is a bug or if I'm missing something in the type checking. Looks good besides this. 👍

Reply:
We talked about this on #synapse-dev, I believe I just missed the `async` on the function. My bad!